src/Umbraco.Web/Editors/ContentController.cs

```diff
@@ -536,7 +540,16 @@ public ContentItemDisplay PostSaveBlueprint([ModelBinder(typeof(ContentItemBinde
         [OutgoingEditorModelEvent]
         public ContentItemDisplay PostSave([ModelBinder(typeof(ContentItemBinder))] ContentItemSave contentItem)
         {
-            return PostSaveInternal(contentItem, content => Services.ContentService.Save(contentItem.PersistedContent, Security.CurrentUser.Id));
+            var contentItemDisplay = PostSaveInternal(contentItem, content => Services.ContentService.Save(contentItem.PersistedContent, Security.CurrentUser.Id));
+            //ensure the active language is still selected
+            if (contentItem.LanguageId.HasValue)

```

Comment:
The model that is returned after the save is used to re-initialize the editor - we do this with all of our editors. Therefore, the model on the client side is replaced by the model returned from the server side and we have to maintain the currently selected language which we know because that is the language that is sent to the server to be saved.